packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx

```diff
@@ -82,6 +90,71 @@ const WorkspaceHeader: React.FC = () => {
 
   const [versionAlert, setVersionAlert] = useState(false);
 
+  const [upgradeButtonData, setUpgradeButtonData] = useState<{
+    trialDaysLeft?: number;
+    trialLeftProgress?: number;
+    showUpgradeTrialButton: boolean;
+    showUpgradeDefaultButton: boolean;
+  }>({ showUpgradeTrialButton: false, showUpgradeDefaultButton: true });
+
+  const getCurrentSubscription = useCallback(async () => {
+    if (currentWorkspace) {
+      await stigg.setCustomerId(currentWorkspace.id);
+      const [subscription] = await stigg.getActiveSubscriptions();
+
+      if (!subscription) {
+        console.error("No active subscription found");
+        return;
+      }
+
+      const DAYS_TO_SHOW_VERSION_ALERT_SINCE_END_OF_TRIAL = 14;
+
+      if (
```

Comment:
please try to use hash map instead of if/else soup

Reply:
Any reason to prefer an hash map here? Try to understand the benefit as it will be less performant to use hash map for these 3 if cases.
_The hashmap has to do more commands to get you there. Fetch hashcode, calculate distance/position, fetch the array entry, and run an equals operation._